packages/rehype-shiki/src/__tests__/plugin.test.mjs

```diff
@@ -55,13 +58,336 @@ describe('rehypeShikiji', async () => {
-      }
-    });
+    rehypeShikiji()(treeToTransform);
 
-    rehypeShikiji()(mockTree);
-    assert.ok(parent.children.some(child => child.tagName === 'CodeTabs'));
+    assert.strictEqual(
+      treeToTransform.children.length,
+      2,
+      'Should not group non-CJS/ESM blocks'
+    );
+    assert.ok(
+      treeToTransform.children.every(child => child.tagName === 'pre'),
+      'All children should remain pre elements'
+    );
+    // Ensure no CodeTabs were created
+    assert.ok(
+      !treeToTransform.children.some(child => child.tagName === 'CodeTabs'),
+      'CodeTabs should not be created for JS/TS'
+    );
+  });
+
+  it('If there are a sequence of codeblock of CJS/ESM, it should create pairs of CodeTabs', () => {
+    const parent = {
+      type: 'root',
+      children: [
+        {
+          tagName: 'pre',
+          children: [
+            {
+              tagName: 'code',
+              data: { meta: 'displayName="CJS"' },
+              properties: { className: ['language-cjs'] },
+            },
+          ],
+        },
+        {
+          tagName: 'pre',
+          children: [
+            {
+              tagName: 'code',
+              data: { meta: 'displayName="ESM"' },
+              properties: { className: ['language-esm'] },
+            },
+          ],
+        },
+        {
+          tagName: 'pre',
+          children: [
+            {
+              tagName: 'code',
+              data: { meta: 'displayName="CJS"' },
+              properties: { className: ['language-cjs'] },
+            },
+          ],
+        },
+        {
+          tagName: 'pre',
+          children: [
+            {
+              tagName: 'code',
+              data: { meta: 'displayName="ESM"' },
+              properties: { className: ['language-esm'] },
+            },
+          ],
+        },
+      ],
+    };
+
+    rehypeShikiji()(parent);
+
+    assert.strictEqual(
+      parent.children.length,
+      2,
+      'Should create two CodeTabs groups'
+    );
+
+    // Check first CodeTabs group
+    const firstGroup = parent.children[0];
+    assert.strictEqual(
+      firstGroup.tagName,
+      'CodeTabs',
+      'Group 1 should be CodeTabs'
+    );
+    assert.strictEqual(
+      firstGroup.properties.languages,
+      'cjs|esm',
+      'Group 1 languages should be cjs|esm'
+    );
+    assert.strictEqual(
+      firstGroup.properties.displayNames,
+      'CJS|ESM',
+      'Group 1 displayNames should be CJS|ESM'
+    );
+    assert.ok(
+      Array.isArray(firstGroup.children) && firstGroup.children.length === 2,
+      'Group 1 should contain 2 code blocks'
+    );
+    assert.strictEqual(
+      firstGroup.children[0].children[0].properties.className[0],
+      'language-cjs',
+      'Group 1, Block 1 should be CJS'
+    );
+    assert.strictEqual(
+      firstGroup.children[1].children[0].properties.className[0],
+      'language-esm',
+      'Group 1, Block 2 should be ESM'
+    );
```

Comment:
Can we use `deepStrictEqual` or `partialDeepStrictEqual` instead of comparing each with it's own assert?